solid/src/main/java/com/inrupt/client/solid/HttpStatus.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.inrupt.client.solid;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+enum HttpStatus {
+    BAD_REQUEST(BadRequestException.STATUS_CODE, "Bad Request"),
```

Comment:
Agreed, that's what I wanted to do initially, but then the `SolidClientException::handle` `switch` statement complains that a union member variable (e.g. `HttpStatus.BAD_REQUEST.code`) is not a constant. I preferred to keep that switch statement rather than a large `if` for readability, but I don't have a strong opinion one way or another.

Reply:
Done in d56823827d5a8a26d9aa6976fa44ab18c030ec27